docs/content/docs/dev/table/sourcesSinks.md

```diff
@@ -197,10 +197,16 @@ A table source can implement further ability interfaces such as `SupportsProject
 mutate an instance during planning. All abilities can be found in the `org.apache.flink.table.connector.source.abilities`
 package and are listed in the [source abilities table](#source-abilities).
 
-The runtime implementation of a `ScanTableSource` must produce internal data structures. Thus, records
-must be emitted as `org.apache.flink.table.data.RowData`. The framework provides runtime converters such
+The returned _scan runtime provider_ provides the runtime implementation for reading the data. There are
+different interfaces for runtime implementation, among which `SourceProvider` is the recommended core interface.
+
+Independent of the provider interface, the source runtime implementation must produce internal data structures.
+Thus, records must be emitted as `org.apache.flink.table.data.RowData`. The framework provides runtime converters such
 that a source can still work on common data structures and perform a conversion at the end.
 
+To support parallelism setting, the dynamic table factory should support the optional `scan.parallelism` option
```

Comment:
Is it worth to mention that `scan.parallelism` has been defined in `FactoryUtil`?

Reply:
Updated to mention it, thanks~